core/src/main/java/org/apache/iceberg/util/ThreadPools.java

```diff
@@ -68,8 +68,9 @@ public static ExecutorService getWorkerPool() {
   /**
    * Return an {@link ExecutorService} that uses the "delete worker" thread-pool.
    *
-   * <p>The size of the delete worker pool limits the number of threads used to compute the
-   * PositionDeleteIndex from the position deletes for a data file.
+   * <p>The size of this worker pool limits the number of tasks concurrently reading delete files
+   * within a single JVM. In most cases, deletes are loaded while reading data on executors. The
```

Comment:
The second sentence is not super useful since we have no guidance on what that size should be. I would probably just drop it, or replace it with "increase this if your executors are  larger?"

In general the whole comment is also very Spark specific ... Maybe just keep it as simple as possible for now? I don't have strong feelings here, except for removing the "has to be big enough" sentence.

Reply:
Agreed, will make it more generic.

Reply:
Made it more generic and added a note about sharing the pool between threads.